### DISCUSSION TITLE
How to reference multiple tables from one id column

### DISCUSSION THREAD

**Discussion by nmfrankel** [prisma/prisma#20621] [Q&A]
### Question

Before submitting my question, I did some research if my question existed. I found the version I implemented, but I'm not satisfied with the answer.

#### Objective
I'm creating an admin platform DB and would like to collect emergency contacts for both `Staff` and `Student` tables.  How can I reference the same `userID` column on the `EmergencyContact` table for both groups of users without worrying about collisions?

In the similar questions they are dealing with `Int` `id`s and don't discuss avoiding collisions, which would definitely exist.

I have attached my schema. Feel free to clarify any details.

#### Similar questions but don't address the issue
https://github.com/prisma/prisma/discussions/19200
https://github.com/prisma/prisma/discussions/11108

### Information about Prisma Schema, Client Queries and Environment (optional)

Here's a barebones version of my schema.prisma 👇

```prisma
model Staff {
  id                String             @id @default(uuid())
  ...
  emergencyContacts EmergencyContact[] @relation("staff_emergency_contact")
}

model Student {
  id                String             @id @default(uuid())
  ...
  emergencyContacts EmergencyContact[] @relation("student_emergency_contact")
}


model EmergencyContact {
  id           String   @id @default(uuid())
  ...
  userType     String
  userId       String
  staff        Staff?   @relation("staff_emergency_contact", fields: [userId], references: [id])
  student      Student? @relation("student_emergency_contact", fields: [userId], references: [id])

  // Maybe utilizing something like this @@index([userType, userId], map: "failsafeID")
}
```

**Comment by janpio** [comment]
Not supported yet. Feature Request: https://github.com/prisma/prisma/issues/1644

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
